Weather App/ViewModel/CurrentWeatherViewModel.swift

```diff
@@ -0,0 +1,113 @@
+            return "thunder"
+        }
+        if conditionText.contains("mist") ||
+            conditionText.contains("fog") {
+            return "mist"
+        }
+        if conditionText.contains("sunny") ||
+            conditionText.contains("clear") ||
+            conditionText.contains("cloud") || conditionText.contains("overcast"){
+            return "clear"
+        }
+        return "default"
+    }
+    
+    func getBackgroundImage(isDay: Int, conditionText: String) -> UIImage {
+        let condition = conditionText.lowercased()
+        var conditionName = checkCondition(conditionText: condition)
+        if conditionName == "default" || conditionName == "thunder" {
+            return UIImage(named: conditionName)!
+        }
+        if isDay == 1 {
+            conditionName += "Day"
+        } else {
+            conditionName += "Night"
+        }
+        return UIImage(named: conditionName)!
+    }
+    
+    func setCurrentLocation () {
+        let locManager = CLLocationManager()
+        locManager.requestWhenInUseAuthorization()
+        var currentLocation: CLLocation!
+
+        if
+           CLLocationManager.authorizationStatus() == .authorizedWhenInUse ||
+           CLLocationManager.authorizationStatus() ==  .authorizedAlways
+        {
+            currentLocation = locManager.location
+            changeLocation(to: "\(currentLocation.coordinate.latitude), \(currentLocation.coordinate.longitude)", isAqi: true)
+        }
+    }
+    
+    func changeLocation (to city: String, isAqi: Bool) {
+        locationName.value = "Loading"
+        ApiFunctions.fetchForecastedData(city: city, days: 11, aqi: isAqi, alerts: false) { data in
+            self.currentWeatherDescription.value = CurrentWeatherCompact(location: data.location.name, temp: WeatherViewModel.roundTwoDecimalLabel(value: data.current.tempC), condition: data.current.condition.text, lat: WeatherViewModel.roundTwoDecimalLabel(value: data.location.lat), long: WeatherViewModel.roundTwoDecimalLabel(value: data.location.lon))
+            self.currentAirQuality.value = AirQualityCompact(co: WeatherViewModel.roundTwoDecimalLabel(value: data.current.airQuality.co), no2: WeatherViewModel.roundTwoDecimalLabel(value: data.current.airQuality.no2), pm25: WeatherViewModel.roundTwoDecimalLabel(value: data.current.airQuality.pm25), pm10: WeatherViewModel.roundTwoDecimalLabel(value: data.current.airQuality.pm10))
+            self.currentWindDescription.value = WindCompact(speed: WeatherViewModel.roundTwoDecimalLabel(value: data.current.windMph), degree: WeatherViewModel.roundTwoDecimalLabel(value: Double(data.current.windDegree)), direction: data.current.windDir)
+            self.backGroundImage.value = self.getBackgroundImage(isDay: data.current.isDay, conditionText: data.current.condition.text)
+            var tempForecastedData = [DayForecastCompact]()
+            for i in 1..<data.forecast.forecastday.count {
```

Comment:
First index stores the data for current data that is the forecasted weather for today,
We are leaving it since in table we are showing forecasted data for coming days